controller/k8s/api.go

```diff
@@ -49,46 +51,47 @@ func NewLister(k8sClient kubernetes.Interface) *Lister {
 	replicationControllerInformer := sharedInformers.Core().V1().ReplicationControllers()
 	serviceInformer := sharedInformers.Core().V1().Services()
 
-	lister := &Lister{
-		NS:     namespaceInformer.Lister(),
-		Deploy: deployInformer.Lister(),
-		RS:     replicaSetInformer.Lister(),
-		Pod:    podInformer.Lister(),
-		RC:     replicationControllerInformer.Lister(),
-		Svc:    serviceInformer.Lister(),
+	api := &API{
+		NS:     namespaceInformer,
+		Deploy: deployInformer,
+		RS:     replicaSetInformer,
+		Pod:    podInformer,
+		RC:     replicationControllerInformer,
+		Svc:    serviceInformer,
 
 		nsSynced:     namespaceInformer.Informer().HasSynced,
 		deploySynced: deployInformer.Informer().HasSynced,
 		rsSynced:     replicaSetInformer.Informer().HasSynced,
 		podSynced:    podInformer.Informer().HasSynced,
 		rcSynced:     replicationControllerInformer.Informer().HasSynced,
 		svcSynced:    serviceInformer.Informer().HasSynced,
-	}
 
-	// this must be called after the Lister() methods
-	sharedInformers.Start(nil)
+		sharedInformers: sharedInformers,
```

Comment:
since the clients of `k8s.API` are now calling `Lister()` themselves, was there any issue with this happening after calls to `sharedInformers.Start(nil)` ?

Reply:
As far as I can tell, all of the clients of `k8s.API` only call `Lister()` in the request path, so the behavior should be unchanged, assuming the process calls `Sync()` before serving requests. This totally relates to #1013, however. It's definitely the case that some or our controller components can serve requests before the caches are fully synced, and we should fix this with readiness probes that fail until caches are synced.